interfaces/cython/cantera/test/test_kinetics.py

```diff
@@ -1272,3 +1397,65 @@ def test_motz_wise(self):
 
         # M&W toggled on (locally) for reaction 9
         self.assertNear(2.0 * k1[9], k2[9]) # sticking coefficient = 1.0
+
+    def test_modify_BMinterface(self):
+        gas = ct.Solution('gri30.yaml')
+        gas.TPX = 300, ct.one_atm, {"CH4": 0.095, "O2": 0.21, "AR": 0.79}
+        surf = ct.Interface('BM_test.yaml', 'Pt_surf', [gas])
+        surf.coverages = 'O(S):0.1, PT(S):0.5, H(S):0.4'
+        gas.TP = surf.TP
+
+        R = surf.reaction(0)
+        R.coverage_deps = {'O(S)': (0.0, 0.0, -3e6)}
+        surf.modify_reaction(0, R)
+
+        # Rate constant should now be independent of H(S) coverage, but
+        # dependent on O(S) coverage
+        k1 = surf.forward_rate_constants[0]
+        surf.coverages = 'O(S):0.2, PT(S):0.4, H(S):0.4'
+        k2 = surf.forward_rate_constants[0]
+        surf.coverages = 'O(S):0.2, PT(S):0.6, H(S):0.2'
+        k3 = surf.forward_rate_constants[0]
+        self.assertNotAlmostEqual(k1, k2)
```

Comment:
Is there a way to use the theory to figure out what their ratio should be so that we could use an `assertNear` test, here?  I'll admit I'm not familiar with how `assertNotAlmostEqual` works, but my naive sense is that it is not a equality test of any sort.  Apologies if I'm wrong, there...

Reply:
According to the [python doc](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertNotAlmostEqual) , `assertNotAlmostEqual` compares the difference between 2 numbers which are rounded to 7 decimal places by default to zero, the assertion is true if the difference is not approximately equal to zero. I am not sure about how to modify the code here because I basically just copied `test_modify_interface`.

Reply:
I think @decaluwe's point is that it's possible to know how different k1 and k2 should be, so you can do a much more meaningful test than just checking that the numbers aren't the same. In this particular case, they should differ by the ratio `exp( delta_theta * E_k / RT )`.